test-in-qmlscene.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+PS3='
+Test a watchface in qmlscene.
+Enter watchface number.
+Quit with any other input -> '
+
+unset options i
+while IFS= read -r -d $'\0' f; do
+  options[i++]="$f"
+done < <(find */ -maxdepth 0 -type d -print0 )
+
+if [ "$1" = "-use12h" ]
+then
+    use12=true
+else
+    use12=false
+fi
+
+if ! command -v qmlscene &> /dev/null
+then
+    echo "$(tput setaf 1)qmlscene could not be found. Install qtcreator from your package manager.$(tput sgr0)"
+    exit
+else
+    echo "$(tput setaf 2)qmlscene found, proceeding...$(tput sgr0)
+         "
+fi
+
+select opt in "${options[@]}"
+
+do
+    if [ -e $opt/usr/share/ ]
+    then
+          sed '/Item {/a height: 640; width: 640; Image { source: "background.jpg"; width: 640; height: 640;} property bool displayAmbient: false' <$opt/usr/share/asteroid-launcher/watchfaces/${opt::-1}.qml >$opt/usr/share/asteroid-launcher/watchfaces/${opt::-1}_scene.qml
+          sed -i "/Item {/a Item { property var value: $use12; id: use12H;} Item { property var time: new Date(); id: wallClock; Timer { interval: 1000; running: true; repeat: true; onTriggered: wallClock.time = new Date(); } }" $opt/usr/share/asteroid-launcher/watchfaces/${opt::-1}_scene.qml
```

Comment:
Instead of modifying the watchface itself have you considered having one QML template file where all of this is hard coded and where a Loader component loads the watchface? You could leave clear placeholders in the QML template for example `@@@WATCHFACE@@@` and `@@@WALLPAPER@@@` then you could copy this template in /tmp and only replace these two placeholders

Reply:
Thank you!
I managed to implement a separate watchfaceloader.qml that contains all the items i previously sed into the <watchface>_scene.qml. Then successfully load the specified <watchface> from this new template/loader file.
I sed replace the watchface path and 12h value using your suggested @@@ flagged placeholders.
That works nicely and is by far better readable and extendable. 

But the mktemp part keeps me stuck.
I can do a temp using `path=$(mktemp /tmp/watchfaceloader.qmlXXXX)` and do sed command in the `$path` referenced temp file. But when i then execute the temp file using the loader it is executed in the /tmp folder and thus the paths to the background image and watchface are wrong since they do not exist in /tmp.
Shall i copy all those to /tmp or is there another solution?